lib/mbaasRequest/mbaasRequest.js

```diff
@@ -177,7 +177,7 @@ function responseHandler(params, cb) {
     if (err || (httpResponse && httpResponse.statusCode >= 400)) {
 
       //There is a specific error message if an environment is unreachable
-      if (httpResponse && httpResponse.statusCode) {
+      if (httpResponse && (httpResponse.statusCode || err.code)) {
```

Comment:
The `httpResponse.statusCode` that comes back from the mbaas for `"ENOTFOUND"/"ETIMEDOUT"` is undefined.  So it then resorts to the default error message and won't show the ."MBaaS environment is not reachable.." message.